File: AngularPWA/src/app/logic/coffee.ts
import { TastingRating } from "./tasting-rating";
import { PlaceLocation } from "./place-location";

export class Coffee {

    // Properties
    type: string;
    rating: number;
    notes: string;
    tastingRating: TastingRating;

    constructor(public name: string = "", public place: string = "", public location: PlaceLocation = null) {
        this.location = new PlaceLocation();
        this.tastingRating = new TastingRating();
    }

}

File: AngularPWA/Models/Coffee.cs
namespace AngularPWA.Models
{
  public class Coffee
  {
    public Coffee(int id, string name, string place, PlaceLocation location)
    {
      Id = id;
      Name = name;
      Place = place;
      Location = location;
    }

    public int Id { get; set; }

    public string Name { get; set; }

    public string Place { get; set; }

    public PlaceLocation Location { get; set; }
  }
}
